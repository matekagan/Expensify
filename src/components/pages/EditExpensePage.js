import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../presentational/ExpenseForm';
import { editExpense, startRemoveExpense } from '../../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense({ id: this.props.expense.id, updates: expense });
        this.props.history.push('/');
    };

    onButtonClick = () => {
        this.props.removeExpense(this.props.expense.id);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <h1>Edit Expense</h1>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    type="button"
                    onClick={this.onButtonClick}
                >
                    Remove
                </button>
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => ({
    editExpense: ({ id, updates }) => dispatch(editExpense({ id, updates })),
    removeExpense: id => dispatch(startRemoveExpense({ id }))
});

const mapStateToProps = (state, props) => ({
    ...props,
    expense: state.expenses.find(expense => expense.id === props.match.params.id)
});


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
