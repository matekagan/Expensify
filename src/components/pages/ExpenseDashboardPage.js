import React from 'react';
import ExpensesList from '../presentational/ExpenseList';
import ExpenseListFilters from '../presentational/ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <h3>Dashboard</h3>
        <p>some text</p>
        <ExpenseListFilters />
        <ExpensesList />
    </div>
);

export default ExpenseDashboardPage;
