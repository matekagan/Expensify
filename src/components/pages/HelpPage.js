import React from 'react';

const HelpPage = () => (
    <div>
        <h1>Help</h1>
        <p>some text number 4</p>
    </div>
);

export default HelpPage;
