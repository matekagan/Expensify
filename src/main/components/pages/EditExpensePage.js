import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../presentational/ExpenseForm';
import { editExpense, startRemoveExpense, startEditExpense } from '../../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense({ id: this.props.expense.id, updates: expense });
        this.props.history.push('/');
    };

    onButtonClick = () => {
        this.props.removeExpense(this.props.expense.id);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button
                        className="button button--secondary"
                        type="button"
                        onClick={this.onButtonClick}
                    >
                        Remove Expense
                    </button>
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => ({
    editExpense: ({ id, updates }) => dispatch(startEditExpense({ id, updates })),
    removeExpense: id => dispatch(startRemoveExpense({ id }))
});

const mapStateToProps = (state, props) => ({
    ...props,
    expense: state.expenses.find(expense => expense.id === props.match.params.id)
});


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
