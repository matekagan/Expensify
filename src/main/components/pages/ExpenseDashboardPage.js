import React from 'react';
import ExpensesList from '../presentational/ExpenseList';
import ExpenseListsFilters from '../presentational/ExpenseListFilters';
import ExpensesSummary from '../presentational/ExpenseSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListsFilters />
        <ExpensesList />
    </div>
);

export default ExpenseDashboardPage;
