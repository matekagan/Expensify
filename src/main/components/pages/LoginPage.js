import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../../actions/auth';

export const LoginPage = ({ logIn }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It is time to get your expenses under control</p>
            <button className="button" type="button" onClick={logIn}> Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = dispatch => ({
    logIn: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
