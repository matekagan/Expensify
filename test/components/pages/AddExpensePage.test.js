import React from 'react';
import { shallow } from 'enzyme';
import { AddExpensePage } from '../../../src/components/pages/AddExpensePage';
import testExpenses from '../../fixtures/expenses';


let onSubmit;
let history;
let wrapper;

beforeEach(() => {
    onSubmit = jest.fn();
    history = {
        push: jest.fn()
    };
    wrapper = shallow(<AddExpensePage addExpense={onSubmit} history={history} />);
});

test('Should render AddExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle onSubmit', () => {
    const expense = testExpenses[1];
    wrapper.find('ExpenseForm').prop('onSubmit')(expense);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(expense);
});
