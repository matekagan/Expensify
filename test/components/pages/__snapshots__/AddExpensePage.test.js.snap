// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddExpensePage correctly 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <p>
    some text number 2
  </p>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
